stl/inc/atomic

```diff
@@ -73,46 +53,174 @@ extern "C" _NODISCARD char __stdcall __std_atomic_has_cmpxchg16b() noexcept;
 #define _ATOMIC_HAS_DCAS 0
 #endif // _STD_ATOMIC_ALWAYS_USE_CMPXCHG16B == 1 || !defined(_M_X64) || defined(_M_ARM64EC)
 
+#define ATOMIC_BOOL_LOCK_FREE 2
+#define ATOMIC_CHAR_LOCK_FREE 2
+#ifdef __cpp_lib_char8_t
+#define ATOMIC_CHAR8_T_LOCK_FREE 2
+#endif // __cpp_lib_char8_t
+#define ATOMIC_CHAR16_T_LOCK_FREE 2
+#define ATOMIC_CHAR32_T_LOCK_FREE 2
+#define ATOMIC_WCHAR_T_LOCK_FREE  2
+#define ATOMIC_SHORT_LOCK_FREE    2
+#define ATOMIC_INT_LOCK_FREE      2
+#define ATOMIC_LONG_LOCK_FREE     2
+#define ATOMIC_LLONG_LOCK_FREE    2
+#define ATOMIC_POINTER_LOCK_FREE  2
+
+// The following code is SHARED with vcruntime and any updates
+// should be mirrored. Also: if any macros are added they should be
+// #undefed in vcruntime as well
+
+enum {
+    _Atomic_memory_order_relaxed,
+    _Atomic_memory_order_consume,
+    _Atomic_memory_order_acquire,
+    _Atomic_memory_order_release,
+    _Atomic_memory_order_acq_rel,
+    _Atomic_memory_order_seq_cst,
+};
+
+#ifndef _INVALID_MEMORY_ORDER
+#ifdef _DEBUG
+#define _INVALID_MEMORY_ORDER _STL_REPORT_ERROR("Invalid memory order")
+#else // ^^^ _DEBUG / !_DEBUG vvv
+#define _INVALID_MEMORY_ORDER
+#endif // _DEBUG
+#endif // _INVALID_MEMORY_ORDER
+
+extern "C" inline void _Check_memory_order(const int _Order) {

```

Comment:
it shouldn't with /EHsc (because /EHc makes /EHs assume that extern "C" functions are noexcept).

We could annotated it with a macro so things work in C mode as well as C++.

To be honest this decl needs a macro anyway to get rid of extern "C" in C mode.